test/Constraints/rdar82828226.swift

```diff
@@ -0,0 +1,8 @@
+// RUN: %target-typecheck-verify-swift
+
+import Foundation
+
+func f(_ n: NSNumber) {}
+let int: Int = 0
+f(int) // expected-error {{cannot convert value of type 'Int' to expected argument type 'NSNumber'}} {{3-3=NSNumber(value: }}
```

Comment:
You'd have to adjust this test case to make sure that `as NSNumber` is used instead of `NSNumber(value:)`.
Also please add some more test cases e.g. using `NSInteger` and as an argument to an operator (you could add a custom one which accepts (NSNumber, NSNumber) to make that easier).